File: SharePreferences/app/src/main/java/com/hfad/sharepreferences/SecondActivity.java
package com.hfad.sharepreferences;

import android.content.SharedPreferences;
import android.os.Bundle;
import android.preference.PreferenceManager;
import android.widget.EditText;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

public class SecondActivity extends AppCompatActivity {
    private static final String TAG = "SecondActivity";
    private TextView mName;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.second_layout);
        mName = (EditText) findViewById(R.id.etName_SL);

        SharedPreferences mPreferences = PreferenceManager.getDefaultSharedPreferences(this);
        SharedPreferences.Editor editor = mPreferences.edit();

       String name = mPreferences.getString(getString(R.string.name), "");
       mName.setText(name);

    }
}

File: LesTwo/app/src/main/java/com/hfad/lestwo/SecondActivity.kt
package com.hfad.lestwo

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.TextView

class SecondActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_second)
        var tvMessage = findViewById<TextView>(R.id.tvMessage)
        var i = intent
        if (i != null) tvMessage.setText(i.getCharSequenceExtra("salt"))
    }
}